core/src/main/java/org/elasticsearch/common/geo/SpatialStrategy.java

```diff
@@ -18,22 +18,53 @@
  */
 package org.elasticsearch.common.geo;
 
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
 
 /**
  *
  */
-public enum SpatialStrategy {
+public enum SpatialStrategy implements Writeable<SpatialStrategy> {
 
-    TERM("term"),
-    RECURSIVE("recursive");
+    TERM((byte) 0, "term"), RECURSIVE((byte) 1, "recursive");
 
+    private final byte id;
     private final String strategyName;
 
-    private SpatialStrategy(String strategyName) {
+    private SpatialStrategy(byte id, String strategyName) {
+        this.id = id;
         this.strategyName = strategyName;
     }
 
     public String getStrategyName() {
         return strategyName;
     }
+
+    @Override
+    public SpatialStrategy readFrom(StreamInput in) throws IOException {
+        byte id = in.readByte();
+        for (SpatialStrategy value : values()) {
```

Comment:
can we use a switch statement here maybe to read and write? like
```JAVA

switch(id) {
  case 0:
    return TERM;
  case 1:
    return RECURSIVE;
}
```

and on writing we can do:
```JAVA
switch(this) {
  case TERM:
    out.writeVint(0);
    break;
  case RECURSIVE:
    out.writeVint(1);
    break;

}
```